presto-native-execution/presto_cpp/main/connectors/PrestoToVeloxConnector.cpp

```diff
@@ -1139,27 +1368,46 @@ HivePrestoToVeloxConnector::toVeloxSplit(
   if (hiveSplit->tableBucketNumber) {
     infoColumns["$bucket"] = std::to_string(*hiveSplit->tableBucketNumber);
   }
-  auto veloxSplit =
-      std::make_unique<velox::connector::hive::HiveConnectorSplit>(
-          catalogId,
-          hiveSplit->fileSplit.path,
-          toVeloxFileFormat(hiveSplit->storage.storageFormat),
-          hiveSplit->fileSplit.start,
-          hiveSplit->fileSplit.length,
-          partitionKeys,
-          hiveSplit->tableBucketNumber
-              ? std::optional<int>(*hiveSplit->tableBucketNumber)
-              : std::nullopt,
-          customSplitInfo,
-          extraFileInfo,
-          serdeParameters,
-          hiveSplit->splitWeight,
-          splitContext->cacheable,
-          infoColumns);
-  if (hiveSplit->bucketConversion) {
-    VELOX_CHECK_NOT_NULL(hiveSplit->tableBucketNumber);
-    veloxSplit->bucketConversion =
-        toVeloxBucketConversion(*hiveSplit->bucketConversion);
+  std::unique_ptr<velox::connector::ConnectorSplit> veloxSplit;
+  if (canUseCudfTableScan()) {
+#ifdef PRESTO_ENABLE_CUDF
+    std::string realPath;
+    if (boost::algorithm::starts_with(hiveSplit->fileSplit.path, "file:")) {
```

Comment:
Not sure which C++ version is being used here but if [C++20](https://en.cppreference.com/w/cpp/string/basic_string/starts_with) or later, you can just use: `if (hiveSplit->fileSplit.path.starts_with("file:"))`